robot_framework/eflyt.py

```diff
@@ -0,0 +1,338 @@
+
+    # Remove header
+    rows.pop(0)
+
+    return len(rows)
+
+
+def get_room_count(browser: webdriver.Chrome) -> int:
+    """Get the number of rooms on the address.
+
+    Args:
+        browser: The webdriver browser object.
+
+    Returns:
+        The number of rooms on the address.
+    """
+    area_room_text = browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_stcPersonTab6_lblAreaText").text
+    room_text = area_room_text.split("/")[1]
+    return int(room_text)
+
+
+def get_applicants(browser: webdriver.Chrome) -> list[str]:
+    """Get a list of applicants' cpr numbers from the applicant table.
+
+    Args:
+        browser: The webdriver browser object.
+
+    Returns:
+        A list of cpr numbers.
+    """
+    table = browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_GridViewMovingPersons")
+    rows = table.find_elements(By.TAG_NAME, "tr")
+
+    # Remove header row
+    rows.pop(0)
+
+    cpr_list = []
+
+    for row in rows:
+        cpr = row.find_element(By.XPATH, "td[2]/a[2]").text
+        cpr_list.append(cpr)
+
+    return cpr_list
+
+
+def approve_case(browser: webdriver.Chrome):
+    """Approve the case and add a note about it.
+
+    Args:
+        browser: _description_
+    """
+    change_tab(browser, 0)
+
+    # Set note
+    create_note(browser, f"{date.today()} Besked fra Robot: Automatisk godkendt.")
+
+    # Click 'Godkend'
+    browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_stcPersonTab1_btnGodkend").click()
+
+    # Click 'OK' in popup
+    browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_stcPersonTab1_btnApproveYes").click()
+
+    # Click 'Godkend' personer
+    browser.find_element(By.ID, "ctl00_ContentPlaceHolder2_ptFanePerson_stcPersonTab1_btnGodkendAlle").click()
+
+
+def get_age(cpr: str) -> int:
+    """Get the age of a person based on their cpr number
+    assuming they are between 0-99 years old.
+
+    Args:
+        cpr: The cpr number in the format 'ddmmyy-xxxx' or 'ddmmyyxxxx'.
+
+    Returns:
+        The age based on the cpr number.
+    """
+    day = int(cpr[0:2])
+    month = int(cpr[2:4])
+    year = int(cpr[4:6]) + 2000
+
+    birthdate = date(year, month, day)
+    current_date = date.today()
+
+    # If the birthdate is in the future revert it back 100 years
+    if birthdate > current_date:
+        birthdate = date(year-100, month, day)
+
+    age = current_date.year - birthdate.year - ((current_date.month, current_date.day) < (birthdate.month, birthdate.day))
+
+    return age
+
+
+def open_case(browser: webdriver.Chrome, case_number: str):
+    """Open a case by searching for it's case number.
+
+    Args:
+        browser: The webdriver browser object.
+        case: The case to open.
+    """
+    # The id for both the search field and search button changes based on the current view hence the weird selectors.
+    case_input = browser.find_element(By.XPATH, '//input[contains(@id, "earchControl_txtSagNr")]')
```

Comment:
Is it due to a typo somewhere that you're only looking for “earch” and not “search”?

Reply:
The id of the search field and button changes from "SearchControl..." to "searchControl..." depending on the state of the site. Otherwise the ids are the same. Therefore I'm just looking for "earchControl..."

Reply:
I could just look for the two seperate ids, but I felt this way makes the code easier to follow